gateways.json

```diff
@@ -29,12 +29,11 @@
 	"https://permaweb.io/ipfs/:hash",
 	"https://ipfs.stibarc.com/ipfs/:hash",
 	"https://ipfs.best-practice.se/ipfs/:hash",
-	"https://:hash.ipfs.2read.net",
+	"https://2read.net/ipfs/:hash",
 	"https://ipfs.2read.net/ipfs/:hash",
 	"https://storjipfs-gateway.com/ipfs/:hash",
 	"https://ipfs.runfission.com/ipfs/:hash",
 	"https://trusti.id/ipfs/:hash",
-	"https://:hash.ipfs.cosmos-ink.net",
```

Comment:
@LinusCDE I'm removing your gateway for now to reduce confusion, but we can re-add it when it supports https://docs.ipfs.io/how-to/address-ipfs-on-web/#automatic-leverage-the-gateway-in-go-ipfs

Reply:
Good idea. Something came up, so it's good to remove it for now. I'll try to address it tomorrow.
Would a redirect from `https://cosmos-ink.net/ipfs/<cid>` and `https://ipfs.cosmos-ink.net/ipfs/<cid>` to `https://<cid>.ipfscosmos-ink.net/` to the trick?

What about the IPNS? Same for that?

Reply:
Yes, redirects will work (ideally, redirect returned by go-ipfs, because that gives you CID conversion for free)

For ipns you use: `https://<libp2p-key>.ipns.cosmos-ink.net/`

Reply:
@lidel I've now added a rule that should fix that.
```
if ($request_uri ~ ^/(?<prefix>ipfs|ipns)/(?<full_path>(?<cid_1>.*?)(?=/)(?<cid_1_path>/?.*)|(?<cid_2>.*))$) {
    # Either $cid_1 or $cid_2 will be empty. When $cid_1 is used, $cid_1_path is not empty as well
    return 301 "https://$cid_1$cid_2.$prefix.cosmos-ink.net$cid_1_path";
  }
```
This nginx rule is applied to `ipfs.cosmos-ink.net`, `ipns.cosmos-ink.net`,  `cosmos-ink.net`.  
Basicially for any path like `https://(ipfs.)cosmos-ink.net/ipfs/<cid>/<otherstuff>` I 301 it to `https://<cid>.ipfs.cosmos-ink.net/<otherstuff>`. Same goes for ipns.

I also added your PublicGateway suggestion from your second mail.

> ideally, redirect returned by go-ipfs, because that gives you CID conversion for free

Running it on my Pi (cosmos-ink.net main domain) would be probably more trouble at this point, because I would need to rebuild the docker-image for armv7 or aarch64 support. But for any conversion that isn't `https://(ipfs|ipns).cosmos-ink.net/(ipfs/ipns)/...`, go-ipfs can feel free to convert that.

For the ipfs/ipns subdomains I also added two entries to prevent using that redirection rule when the cid is already in the subdomain (to allow `https://<cid>.ipfs.cosmos-ink.net/ipfs/...` in case ipfs web apps have a directory called "ipfs").

Could you check whether it now works as expected? I'll probably then create a PR to re-add the entry.

Reply:
It works fine for CIDv1, but redirect alone is not enough if someone passes CIDv0.

To illustrate, if you open https://cosmos-ink.net/ipfs/QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR  it should redirect to https://bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi.ipfs.cosmos-ink.net/
 (go-ipfs takes care of CID conversion, without it CIDv0 will be force-lowercased and fail).

ps. Do you mind commenting in https://github.com/ipfs/go-ipfs/issues/4931 regarding the need for  docker image working out of the box on rpi?